CoreScriptsRoot/CoreScripts/Topbar.lua

```diff
@@ -1253,6 +1262,70 @@ local function Create3DChatIcon(topBarInstance, panel, menubar)
 	return menuItem
 end
 
+local function Create3DUserGuiToggleIcon(topBarInstance, panel, menubar)
+	local menuItem = Create3DMenuItem()
+	menuItem:SetHoverText("2D UI")
+
+	local icon = Util.Create "ImageLabel" {
+		Parent = menuItem:GetInstance(),
+
+		Position = UDim2.new(0.5, -14, 0.5, -14),
+		Size = UDim2.new(0, 28, 0, 28),
+
+		BackgroundTransparency = 1,
+
+		Image = "rbxasset://textures/ui/VR/toggle2D.png"
+	}
+
+	local userGuiPanel = Panel3D.Get("UserGui")
+	userGuiPanel:SetType(Panel3D.Type.Fixed)
+	userGuiPanel:ResizePixels(300, 125)
+	userGuiPanel:SetVisible(false, false)
+
+	menuItem.MouseButton1Click:connect(function()
+		if InputService.VREnabled then
+			userGuiPanel:SetVisible(not userGuiPanel.isVisible, false)
+		end
+	end)
+
+	function userGuiPanel:OnVisibilityChanged(visible)
+		if visible then
+			local headLook = Panel3D.GetHeadLookXZ(true)
+			userGuiPanel.localCF = headLook * CFrame.Angles(math.rad(5), 0, 0) * CFrame.new(0, 0, 5)
+		end
+		menuItem:SetActive(visible)
+		local success, msg = pcall(function()
+			CoreGuiService:SetUserGuiRendering(true, visible and userGuiPanel:GetPart() or nil, Enum.NormalId.Front)
+		end)
+		if not success then
+			print("Topbar - userGuiPanel:OnVisibilityChanged:" , msg)
```

Comment:
It will only print if the pcall fails; the function itself won't fail if the FFlag is off, but will if it isn't in the engine yet.